config/ax_python_devel.m4

```diff
@@ -97,23 +97,13 @@ AC_DEFUN([AX_PYTHON_DEVEL],[
 	# Check for a version of Python >= 2.1.0
 	#
 	AC_MSG_CHECKING([for a version of Python >= '2.1.0'])
-	ac_supports_python_ver=`cat<<EOD | $PYTHON -
-from __future__ import print_function;
-import sys;
-try:
-	from packaging import version;
-except ImportError:
-	from distlib import version;
-ver = sys.version.split ()[[0]];
-(tst_cmp, tst_ver) = ">= '2.1.0'".split ();
-tst_ver = tst_ver.strip ("'");
-eval ("print (version.LegacyVersion (ver)"+ tst_cmp +"version.LegacyVersion (tst_ver))")
-EOD`
+	ac_supports_python_ver=`$PYTHON -c "import sys; \
+		ver = sys.version.split ()[[0]]; \
+		print (ver >= '2.1.0')"`
```

Comment:
I believe this logic has the same flaw this was patched to fix, but because the comparison is specifically with 2.1.0, it doesn't come up.

e.g.
```
>>> a
'3.9.2'
>>> b
'2.1.0'
>>> c
'3.10.0'
>>> a > c
True
>>> c > b
True
```

They're assuming string comparison of versions works like humans would intuitively do the comparison, and that's just not true.

I believe that I specifically used LegacyVersion because there was some reason the non-Legacy one had problems, like it didn't handle things like pre or rc suffixes on versions or something to that effect.

Oh, right, because Version doesn't handle anything that doesn't pass PEP 440, was why. I thought I recalled there being some case in the CI or my own testbeds that got bitten by that, or maybe some distro naming that broke it, and that was why. 

I'm not sure what we'd like to do, tbh - the most "correct" solution would be to use non-Legacy version, or to fallback to non-Legacy if it's absent, but if we don't have a concrete example of why it broke, maybe we don't care to add the additional complication.

(I also remembered why I didn't upstream this - it adds an external dependency, whereas their solution "only" breaks when the string width is inconsistent between versions and has no such dependency.)